File: classlog-frontend/src/app/model/entities/task-dto.ts
import {UserDto} from "./user-dto";

export interface TaskDto {
  id: number; // Task ID
  taskName: string; // Name of the task
  description: string; // Description of the task
  dueDate: Date; // Due date of the task (ISO format string)
  createdAt: Date; // Creation timestamp (ISO format string)
  createdBy: UserDto; // User who created the task
  score: number; // Task score
}

File: classlog-backend/src/main/java/com/example/classlog/dto/TaskDto.java
package com.example.classlog.dto;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TaskDto {

  private Long id;
  private String taskName;
  private String description;
  private LocalDateTime dueDate;
  private LocalDateTime createdAt;
  private UserDto createdBy;
  private Integer score;
}
